src/types/ut_types/UTF8OutPipeReaderTests.cpp

```diff
@@ -0,0 +1,199 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+#include "precomp.h"
+#include "WexTestClass.h"
+#include "..\..\inc\consoletaeftemplates.hpp"
+
+#include "..\inc\UTF8OutPipeReader.hpp"
+
+#include <winrt/Windows.Foundation.h>
+#include <winrt/Windows.Foundation.Collections.h>
+
+using namespace WEX::Common;
+using namespace WEX::Logging;
+using namespace WEX::TestExecution;
+
+class UTF8OutPipeReaderTests
+{
+    TEST_CLASS(UTF8OutPipeReaderTests);
+
+    TEST_METHOD(TestUtf8MergePartials)
+    {
+        // The test uses the character 'GOTHIC LETTER HWAIR' (U+10348) as an example
+        // Its UTF-8 representation consists of four bytes:
+        //   1    2    3    4
+        // 0xF0 0x90 0x8D 0x88
+        //
+        // For the test a std::string is filled with 4102 '.' characters to make sure it exceeds the
+        //  buffer size of 4096 bytes in UTF8OutPipeReader.
+        //
+        // This figure shows how the string is getting changed for the 7 sub-tests. The digits 1 to 4
+        //  represent the four bytes of the 'Hwair' letter. The vertical bar represents the buffer boundary.
+        // Test 1: [more points] . . S 1 2 3 4 T|U V W X Y Z . .
+        // Test 2: [more points] . . S T 1 2 3 4|U V W X Y Z . .
+        // Test 3: [more points] . . S T U 1 2 3|4 V W X Y Z . .
+        // Test 4: [more points] . . S T U V 1 2|3 4 W X Y Z . .
+        // Test 5: [more points] . . S T U V W 1|2 3 4 X Y Z . .
+        // Test 6: [more points] . . S T U V W X|1 2 3 4 Y Z . .
+        // Test 7: [more points] . . S T U V W X|Y 1 2 3 4 Z . .
+        //
+        // Tests 1, 6, and 7 prove proper ASCII handling.
+        // Test 2 leaves all four bytes of 'Hwair' in the first chunk.
+        // Test 3, 4, and 5 move the partials from the end of the first chunk to the begin of the
+        //  second chunk.
+        //
+        // At the beginning of a test the whole string is converted into a winrt::hstring for reference.
+        // During the test a second hstring is concatenated out of the chunks that we get from
+        //  UTF8OutPipeReader::Read. Each chunk is separately converted to hstring in order to make
+        //  sure it would be corrupted if we get UTF-8 partials.
+        // The test is positive if both hstrings are equal.
+
+        const size_t bufferSize{ 4096 }; // NOTE: This has to match the buffer size in UTF8OutPipeReader!
+        int subTestCounter{}; // contains the number of performed sub-tests
+        int successCounter{}; // contains the number of passed sub-tests
+
+        std::string utf8TestString(bufferSize + 8, '.'); // create a test string with the required size
+
+        // Test 1:
+        //                                                           ||
+        utf8TestString.replace(bufferSize - 6, 12, "S\xF0\x90\x8D\x88TUVWXYZ");
+        if (SUCCEEDED(RunTest(utf8TestString)))

```

Comment:
You could just do `VERIFY_SUCCEEDED` for these instead of storing a count of tests and successful tests.

You can call VERIFY multiple times per test. If any VERIFY fails, the whole test fails from TAEF's point of view.

```
        VERIFY_SUCCEEDED(RunTest(utf8TestString));
````